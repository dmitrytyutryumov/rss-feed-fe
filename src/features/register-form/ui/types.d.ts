export type FormValues = {
  email: string
  firstName: string
  secondName: string
  password: string
  passwordConfirmation: string
}
