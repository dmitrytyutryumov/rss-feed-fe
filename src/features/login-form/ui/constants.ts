import * as yup from 'yup'

import { FormValues } from './types'

export const INITIAL_VALUES: FormValues = { email: '', password: '' }

export const VALIDATION_SCHEMA = yup.object().shape({
  email: yup.string().email('Invalid email').required('Required'),
  password: yup
    .string()
    .min(6, 'Too Short!')
    .max(250, 'Too Long!')
    .required('Required'),
})
