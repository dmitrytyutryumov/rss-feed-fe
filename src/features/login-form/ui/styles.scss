@import 'src/app/styles/vars.scss';

.login-form {
  display: grid;
  gap: 1rem;

  &__title {
    font-family: inherit;
    font-size: $fs-800;
    line-height: 1.1;
    font-weight: 700;
  }

  &__wrapper {
    display: grid;
    gap: 1rem
  }

  &__reset-password {
    color: $disabled;
    font-size: $fs-300;
    
    cursor: pointer;
  }

  &__redirect-wrapper {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    gap: 0.5rem;

    font-size: $fs-400;
  }

  &__redirect-text {
    color: $disabled;
  }

  &__redirect-link {
    color: $color-primary;
    text-transform: uppercase;
    font-weight: 600;

    cursor: pointer;
  }
}