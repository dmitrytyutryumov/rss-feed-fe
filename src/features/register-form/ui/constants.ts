import * as yup from 'yup'

import { FormValues } from './types'

export const INITIAL_VALUES: FormValues = {
  email: '',
  firstName: '',
  secondName: '',
  password: '',
  passwordConfirmation: '',
}

export const VALIDATION_SCHEMA = yup.object().shape({
  email: yup.string().email('Invalid email').required('Required'),
  firstName: yup.string(),
  secondName: yup.string(),
  password: yup.string().min(6).max(250).required('Required'),
  passwordConfirmation: yup
    .string()
    .oneOf([yup.ref('password'), null], 'Passwords must match'),
})
