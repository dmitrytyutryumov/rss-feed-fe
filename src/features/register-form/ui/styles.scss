@import 'src/app/styles/vars.scss';

.register-form {
  display: grid;
  gap: .7rem;

  &__title {
    font-family: inherit;
    font-size: $fs-800;
    line-height: 1.1;
    font-weight: 700;
  }

  &__wrapper {
    display: grid;
    gap: 1rem
  }

  &__redirect-wrapper {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    gap: 0.5rem;

    font-size: $fs-400;
  }

  &__redirect-text {
    color: $disabled;
  }

  &__redirect-link{
    color: $color-primary;
    text-transform: capitalize;
    font-weight: 600;

    cursor: pointer;
  }

  &__pair-field-container {
    display: flex;
    flex-wrap: wrap;
    gap: .7rem;
  }

  &__pair-field-container > * {
    flex: 1
  }
}