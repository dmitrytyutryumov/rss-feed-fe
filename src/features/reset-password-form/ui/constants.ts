import * as yup from 'yup'

import { FormValues } from './types'

export const INITIAL_VALUES: FormValues = {
  email: '',
}

export const VALIDATION_SCHEMA = yup.object().shape({
  email: yup.string().email('Invalid email').required('Required'),
})
