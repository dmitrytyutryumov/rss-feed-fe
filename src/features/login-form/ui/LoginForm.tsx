import { Formik, Form, FormikHelpers } from 'formik'
import { Link } from 'react-router-dom'

import { FormInput, SubmitButton } from 'shared/components'
import { INITIAL_VALUES, VALIDATION_SCHEMA } from './constants'
import { FormValues } from './types'

import './styles.scss'

export const LoginForm = () => {
  const submitHandler = async (
    values: FormValues,
    { setSubmitting }: FormikHelpers<FormValues>
  ) => {
    console.log({ values })
    setSubmitting(false)
  }

  return (
    <Formik
      initialValues={INITIAL_VALUES}
      onSubmit={submitHandler}
      validationSchema={VALIDATION_SCHEMA}
    >
      <Form className="login-form">
        <p className="login-form__title">Login</p>
        <div className="login-form__wrapper">
          <FormInput name="email" type="Email" placeholder="Type your email" />
          <FormInput
            name="password"
            type="password"
            placeholder="Type your password"
          />
          <Link className="login-form__reset-password" to="../reset-password">
            Forgot your password?
          </Link>
          <SubmitButton />
        </div>
        <div className="login-form__redirect-wrapper">
          <p className="login-form__redirect-text">
            Don&rsquo;t have an account?
          </p>
          <Link className="login-form__redirect-link" to="../register">
            Create one
          </Link>
        </div>
      </Form>
    </Formik>
  )
}
