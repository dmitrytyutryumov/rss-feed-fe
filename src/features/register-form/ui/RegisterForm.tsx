import { Formik, Form, FormikHelpers } from 'formik'
import { Link } from 'react-router-dom'
import { useDispatch } from 'react-redux'

import { FormInput, SubmitButton } from 'shared/components'
import { IRegisterFormValues, registerUserAction } from 'entities/auth'

import { INITIAL_VALUES, VALIDATION_SCHEMA } from './constants'
import './styles.scss'

export const RegisterForm = () => {
  const dispatch = useDispatch()
  const submitHandler = async (
    values: IRegisterFormValues,
    { setSubmitting }: FormikHelpers<IRegisterFormValues>
  ) => {
    dispatch(registerUserAction(values))
    setSubmitting(false)
  }

  return (
    <Formik
      initialValues={INITIAL_VALUES}
      onSubmit={submitHandler}
      validationSchema={VALIDATION_SCHEMA}
    >
      <Form className="register-form">
        <p className="register-form__title">Create an account</p>
        <div className="register-form__redirect-wrapper">
          <p className="register-form__redirect-text">
            Already have an account?
          </p>
          <Link className="register-form__redirect-link" to="../login">
            Log in
          </Link>
        </div>

        <div className="register-form__wrapper">
          <FormInput name="email" type="Email" placeholder="Email" />
          <div className="register-form__pair-field-container">
            <FormInput name="firstName" type="text" placeholder="First Name" />
            <FormInput
              name="secondName"
              type="text"
              placeholder="Second Name"
            />
          </div>
          <FormInput
            name="password"
            type="password"
            placeholder="Type your password"
          />
          <FormInput
            name="passwordConfirmation"
            type="password"
            placeholder="Repeat password"
          />
          <SubmitButton />
        </div>
      </Form>
    </Formik>
  )
}
