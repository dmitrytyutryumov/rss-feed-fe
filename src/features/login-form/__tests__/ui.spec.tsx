import { render } from '@testing-library/react'
import { useField } from 'formik'
import { BrowserRouter } from 'react-router-dom'

import { LoginForm } from '../ui'

describe('test login form', () => {
  beforeEach(() => {
    useField.mockImplementation(() => [{}, { error: null, touched: false }])
  })
  it('test login form snapshot', () => {
    // Act
    const { asFragment } = render(<LoginForm />, { wrapper: BrowserRouter })
    // Assert
    expect(asFragment()).toMatchSnapshot()
  })
})
