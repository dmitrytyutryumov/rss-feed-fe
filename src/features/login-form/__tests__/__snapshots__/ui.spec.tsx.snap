// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test login form test login form snapshot 1`] = `
<DocumentFragment>
  <form
    action="#"
    class="login-form"
  >
    <p
      class="login-form__title"
    >
      Login
    </p>
    <div
      class="login-form__wrapper"
    >
      <div
        class="form-section undefined"
        data-has-error="false"
      >
        <input
          class="form-section__input"
          placeholder="Type your email"
          type="Email"
        />
      </div>
      <div
        class="form-section undefined"
        data-has-error="false"
      >
        <input
          class="form-section__input"
          placeholder="Type your password"
          type="password"
        />
      </div>
      <a
        class="login-form__reset-password"
        href="/reset-password"
      >
        Forgot your password?
      </a>
      <button
        class="submit-btn undefined"
        type="submit"
      >
        Submit
      </button>
    </div>
    <div
      class="login-form__redirect-wrapper"
    >
      <p
        class="login-form__redirect-text"
      >
        Donâ€™t have an account?
      </p>
      <a
        class="login-form__redirect-link"
        href="/register"
      >
        Create one
      </a>
    </div>
  </form>
</DocumentFragment>
`;
